### PR TITLE
Combine range options, add minimum scale range

### LINKED ISSUE #396: Extreme zoom values crash the plugin
First of all many thanks for this awesome plugin ❤️ During some tests I noticed that if you zoom very far into the graph or out the plugin crashes. This is very easy to reproduce, in the first example of the documentation simply set the zoom speed to 100000 and zoom in for a few seconds.

image

image

As far as I can see the integer is overflowing and the axis disappears.

image

Would it make sense to intercept this? Or could you describe a method to define a maximum zoom level (defining for example just to allow zooming until the values are beyond three decimal places). The minmum zoom level can be set with rangeMin/rangeMax of course. Thanks a lot already!

### PR DESCRIPTION
**Pull request by kurkle**
Resolves #396

### CODE REVIEW

**Review by etimberg — COMMENTED** [review]
I think it makes sense and perhaps closes a few other tickets asking for zoom & pan limits. Is the options change breaking?

**Comment by kurkle** [comment]
| Filename | Size | Change |
| :--- | :---: | :---: |
| `dist/chartjs-plugin-zoom.esm.js` | 5.53 kB | -11 B (0%) |
| `dist/chartjs-plugin-zoom.js` | 5.78 kB | -5 B (0%) |
| `dist/chartjs-plugin-zoom.min.js` | 3.34 kB | -8 B (0%) |

**Comment by kurkle** [comment]
> I think it makes sense and perhaps closes a few other tickets asking for zoom & pan limits. Is the options change breaking?

It is, added the label.

**Review by etimberg — APPROVED** [review]

File: docs/guide/options.md [hunk 1]
```diff
@@ -69,3 +68,22 @@ const chart = new Chart('id', {
...
+## Range
+
+Range options define the limits for scale pan and zoom.
+
+### Range options
+
+| Name | Type | Description
+| ---- | -----| -----------
+| `x` | [`ScaleRange`](#scale-range) | Range limits for x-axis
+| `y` | [`ScaleRange`](#scale-range) | Range limits for x-axis
```
Comment (jledentu) [inline docs/guide/options.md:81]: typo: y-axis

File: docs/guide/options.md [hunk 2]
```diff
@@ -69,3 +68,22 @@ const chart = new Chart('id', {
...
+
+#### Scale Range
+
+| Name | Type | Description
+| ---- | -----| -----------
+| `min` | `number` | Minimun allowed value for scale.min
+| `max` | `number` | Maximum allowed value for scale.max
+| `range` | `number` | Minimum allowed range (max - min). This defines the max zoom level.
```
Comment (jledentu) [inline docs/guide/options.md:89]: Maybe it's just me but `range.range` naming is a bit confusing, and doesn't reflect that defines a limit. ^^ What do you think about `minRange` or `limit`... ?

**Review by jledentu — COMMENTED** [review]
Great! This `range` limit will be very useful! :+1:

File: docs/guide/options.md (hunk 2 above)
Comment (kurkle) [reply docs/guide/options.md:89]: Changed the option key to `limits`, so `limits.min`, `limits.max` and `limits.range`. What do you think?

Comment (kurkle) [reply docs/guide/options.md:89]: Maybe it should still be `minRange` to leave room for possible `maxRange`.

Comment (jledentu) [reply docs/guide/options.md:89]: As you want, I think that `limits.min`, `limits.max` and `limits.range` is nice! Thanks!

**Review by jledentu — APPROVED** [review]

**Review by etimberg — APPROVED** [review]

**Comment by kurkle** [comment]
Last change to `minRange`, so we can have `maxRange` in the future if needed.

**Review by etimberg — APPROVED** [review]

**Pull request merged**